data-plane/core/service/src/bindings/adapter.rs

```diff
@@ -1,96 +1,376 @@
-    app: Arc<App<P, V>>,
+// Re-export uniffi for proc macros
+use uniffi;
+
+// ============================================================================
+// UniFFI Type Definitions
+// ============================================================================
+
+/// Global Tokio runtime for async operations
+static GLOBAL_RUNTIME: OnceLock<Arc<tokio::runtime::Runtime>> = OnceLock::new();
+
+/// Get or initialize the global Tokio runtime
+///
+/// Configured for FFI workloads with:
+/// - Worker threads: 2x CPU cores (to handle blocking operations better)
+/// - Max blocking threads: 512 (allows high concurrency from FFI calls)
+/// - Named threads for easier debugging
+fn get_runtime() -> &'static Arc<tokio::runtime::Runtime> {
+    GLOBAL_RUNTIME.get_or_init(|| {
+        // Calculate optimal worker thread count
+        // Use 2x CPU cores for workloads with blocking operations from FFI
+        let num_workers = std::env::var("SLIM_TOKIO_WORKERS")
+            .ok()
+            .and_then(|s| s.parse().ok())
+            .unwrap_or_else(|| {
+                let cpus = num_cpus::get();
+                (cpus * 2).max(4) // At least 4 workers, preferably 2x CPUs
+            });
+
+        // Allow configurable max blocking threads (default: 512)
+        let max_blocking = std::env::var("SLIM_MAX_BLOCKING_THREADS")
+            .ok()
+            .and_then(|s| s.parse().ok())
+            .unwrap_or(512);
+
+        Arc::new(
+            tokio::runtime::Builder::new_multi_thread()
+                .worker_threads(num_workers)
+                .max_blocking_threads(max_blocking)
+                .thread_name_fn(|| {
+                    static ATOMIC_ID: AtomicUsize = AtomicUsize::new(0);
+                    let id = ATOMIC_ID.fetch_add(1, Ordering::SeqCst);
+                    format!("slim-rt-{}", id)
+                })
+                .enable_all()
+                .build()
+                .expect("Failed to create Tokio runtime"),
+        )
+    })
+}
 
-    /// Channel receiver for notifications from the app
-    notification_rx: Arc<RwLock<mpsc::Receiver<Result<Notification, SessionError>>>>,
+/// Initialize the crypto provider
+#[uniffi::export]
+pub fn initialize_crypto() {
+    // Crypto initialization happens automatically in slim_auth
+    // Also initialize the global runtime
+    let _ = get_runtime();
 }
 
-impl<P, V> BindingsAdapter<P, V>
-where
-    P: TokenProvider + Send + Sync + Clone + 'static,
-    V: Verifier + Send + Sync + Clone + 'static,
-{
-    /// Create a new AppAdapter wrapping the given App
-    pub fn new_with_app(
-        app: App<P, V>,
-        notification_rx: mpsc::Receiver<Result<Notification, SessionError>>,
-    ) -> Self {
-        Self {
-            app: Arc::new(app),
-            notification_rx: Arc::new(RwLock::new(notification_rx)),
+/// Get the version of the SLIM bindings
+#[uniffi::export]
+pub fn get_version() -> String {
+    env!("CARGO_PKG_VERSION").to_string()
```

Comment:
This is the version of the `agntcy-slim-service` crate, we should probably move the uniffi-bindings to a separate crate and give its own version, which will then define the version of each language binding.